server/src/main/java/org/elasticsearch/health/node/DiskHealthIndicatorService.java

```diff
@@ -0,0 +1,284 @@
+            ? HealthStatus.RED
+            : HealthStatus.merge(diskHealthInfoMap.values().stream().map(DiskHealthInfo::healthStatus));
+        if (HealthStatus.GREEN.equals(healthStatus)) {
+            symptom = "Disk usage is within configured thresholds";
+            diagnosisList = List.of();
+        } else {
+            final Set<String> problemNodes;
+            final Set<String> problemIndices;
+            if (HealthStatus.RED.equals(healthStatus)) {
+                problemNodes = diskHealthInfoMap.entrySet()
+                    .stream()
+                    .filter(entry -> HealthStatus.RED.equals(entry.getValue().healthStatus()))
+                    .map(Map.Entry::getKey)
+                    .collect(Collectors.toSet());
+                problemIndices = Stream.concat(getIndicesForNodes(problemNodes).stream(), indicesWithBlock.stream())
+                    .collect(Collectors.toSet());
+                symptom = String.format(
+                    Locale.ROOT,
+                    "%d node%s out of disk space.%s",
+                    problemNodes.size(),
+                    problemNodes.size() > 1 ? "s are" : " is",
+                    problemIndices.size() > 0
+                        ? String.format(
+                            Locale.ROOT,
+                            " As a result %d ind%s cannot process any more updates.",
+                            problemIndices.size(),
+                            problemIndices.size() > 1 ? "ices" : "ex"
+                        )
+                        : ""
+                );
+                if (problemIndices.size() > 1) {
+                    impacts.add(RED_HEALTH_AFFECTED_INDICES_INGEST_IMPACT);
+                }
+            } else {
+                problemNodes = diskHealthInfoMap.entrySet()
+                    .stream()
+                    .filter(entry -> HealthStatus.YELLOW.equals(entry.getValue().healthStatus()))
+                    .map(Map.Entry::getKey)
+                    .collect(Collectors.toSet());
+                problemIndices = getIndicesForNodes(problemNodes);
+                symptom = String.format(
+                    Locale.ROOT,
+                    "%d node%s increased disk space.%s",
+                    problemNodes.size(),
+                    problemNodes.size() > 1 ? "s have" : " has",
+                    problemIndices.size() > 0
+                        ? String.format(
+                            Locale.ROOT,
+                            " As a result %d ind%s at risk of not being able to process any more updates.",
+                            problemIndices.size(),
+                            problemIndices.size() > 1 ? "ices are" : "ex is"
+                        )
+                        : ""
+                );
+                if (problemIndices.size() > 1) {
+                    impacts.add(YELLOW_HEALTH_AFFECTED_INDICES_INGEST_IMPACT);
```

Comment:
I think this is supposed to be `RED_HEALTH_AFFECTED_INDICES_INGEST_IMPACT` since at least one index is blocked.

Reply:
These are nodes that reported `YELLOW` and that didn't have any blocks, so I was intentionally using the softer language. Are you saying you think we ought to use the same language in both cases (RED and YELLOW)?

Reply:
No definitely not, I have probably misunderstood something, I will go through it again.